### ISSUE TITLE
backup works a treat.  not so restore

### ISSUE THREAD

**Issue by pms1969** [stevesloka/ark-blockstore-rook#2]
Backing up seems to work, but when I restore I get errors

```
MountVolume.SetUp failed for volume "pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2" : mount command failed, status: Failure, reason: Rook: Mount volume failed: failed to attach volume replicapool/pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2: failed to map image replicapool/pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2 cluster rook. failed to map image replicapool/pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2: Failed to complete 'rbd': exit status 2. . output: rbd: sysfs write failed In some cases useful info is found in syslog - try "dmesg | tail". rbd: map failed: (2) No such file or directory
```
The pod is stuck at Waiting: ContainerCreation

The PVC is classified as bound and can't be deleted.  Deleting the pod doesn't work either as it comes back with:
```
MountVolume.SetUp failed for volume "pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2" : mount command failed, status: Failure, reason: Rook: Mount volume failed: failed to attach volume pvc-5377f68a-4fb0-11e8-aa2e-0690e07debf2 for pod default/mongo-mongodb-68bfb98d5c-r6cdk. Volume is already attached by pod default/mongo-mongodb-68bfb98d5c-7qrxv. Status Pending
```

Any helpful ideas?

**Comment by pms1969** [comment]
the logs from the rook-api
```
2018/06/13 12:31:10 rook-rest-api is up and running! 2018-06-13 12:31:10.650612063 +0000 UTC m=+0.001835545
2018/06/13 13:03:48 Called initRook()
There was an error creating the volume:  exit status 2
10.2.5.10 - - [13/Jun/2018:13:03:48 +0000] "POST /block/replicapool/replicapool%7C%7Cpvc-0e02e8cd-4f85-11e8-aa2e-0690e07debf2%7C%7C935d41c7-b2b
7-4047-a357-6caa1a211c03/pvc-0e02e8cd-4f85-11e8-aa2e-0690e07debf2 HTTP/1.1" 500 0
```
the initRook() and error continue to repeat

**Comment by pms1969** [comment]
Actually, there is nothing that is big enough in the backup that looks like it could be the pv's.  Assuming something wrong with the backup now.

**Comment by pms1969** [comment]
OK, Discussion on the kubernetes/ark-dr slack channel has enlightned me enough to understand that cross cluster backup/restore is not possible with this plugin.  WIll investitgate the restic integration.

**Comment by stevesloka** [comment]
Hey @pms1969 sorry for the delay in getting back to you! Yes, this repo may not have all the bits for a proper backup of rook. Last time I talked with that team, we chatted about adding better snapshot support into rook, but there were some underlying issues with how ceph could support snapshots. 

I'm interested in the new Resic stuff, I may put together an update to this repo to play with those new bits.

**Comment by pms1969** [comment]
@stevesloka Thanks for getting back to me.  

I gave Restic a go, and it's a good solution as long as the PV's are actively attached to a pod.  However that's not my situation. :(

[I've raised an issue on the Ark project](https://github.com/heptio/ark/issues/556) at the request of one of the team after our discussion on slack.

The upshot is that I may have to have a fiddle with this plugin and try adding S3 persistence to it in some way.  Currently plan to look at [backy2](http://backy2.com/docs/index.html) as a means of achieving this.  I have no experience with it (or ceph for that matter), so it may not be suitable.  Will let you know, and provide a PR if it works out.

**Comment by pms1969** [comment]
@stevesloka I've managed to get an off cluster backup an restore working.  The code is heavily modified from master, but after I've cleaned it up, I'd like to provide a PR if you're willing?

The basis for the backup/restore was to do an `rbd export/import` and then copy the file to/from s3.  I'm not really sure how you'd go about generalising this for other cloud providers, and it's only worked for me once so far, so I'm going to spend a bit of time testing and retesting.  Are you aware of any problems with that approach?
